Doesn't build with GCC 7.1

I first tried the included Makefile, then generating it from the cmake. You can see the log of that below. It looks like you're not telling the compiler to use the right `-std=` or something?

```
✓ fratti@archbook gpuvis $ make -j4
Building _release/gpuvis...
---- src/gpuvis.cpp ----
---- src/gpuvis_graph.cpp ----
---- src/gpuvis_utils.cpp ----
---- src/tdopexpr.cpp ----
In file included from src/gpuvis_graph.cpp:44:0:
src/gpuvis_utils.h:35:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_1;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:36:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_2;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:37:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_3;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:38:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_4;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:39:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_5;
            ^~~~~~~~~~~~
src/gpuvis_graph.cpp: In member function ‘void graph_info_t::init_row_info(TraceWin*, const std::vector<GraphRows::graph_rows_info_t>&)’:
src/gpuvis_graph.cpp:385:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_print_timeline, win, _1 );
                                    ^~~~
src/gpuvis_graph.cpp:385:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_print_timeline, win, _1 );
                                    ^~~~
                                    find
src/gpuvis_graph.cpp:385:87: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_print_timeline, win, _1 );
                                                                                       ^~
src/gpuvis_graph.cpp:385:87: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_print_timeline, win, _1 );
                                                                                       ^~
                                                                                       y1
src/gpuvis_graph.cpp:390:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_plot, win, _1 );
                                    ^~~~
src/gpuvis_graph.cpp:390:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_plot, win, _1 );
                                    ^~~~
                                    find
src/gpuvis_graph.cpp:390:77: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_plot, win, _1 );
                                                                             ^~
src/gpuvis_graph.cpp:390:77: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_plot, win, _1 );
                                                                             ^~
                                                                             y1
src/gpuvis_graph.cpp:395:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_timeline, win, _1 );
                                    ^~~~
src/gpuvis_graph.cpp:395:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_timeline, win, _1 );
                                    ^~~~
                                    find
src/gpuvis_graph.cpp:395:85: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_timeline, win, _1 );
                                                                                     ^~
src/gpuvis_graph.cpp:395:85: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_timeline, win, _1 );
                                                                                     ^~
                                                                                     y1
src/gpuvis_graph.cpp:400:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_hw_row_timeline, win, _1 );
                                    ^~~~
src/gpuvis_graph.cpp:400:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_hw_row_timeline, win, _1 );
                                    ^~~~
                                    find
src/gpuvis_graph.cpp:400:88: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_hw_row_timeline, win, _1 );
                                                                                        ^~
src/gpuvis_graph.cpp:400:88: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_hw_row_timeline, win, _1 );
                                                                                        ^~
                                                                                        y1
src/gpuvis_graph.cpp:405:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_events, win, _1 );
                                    ^~~~
src/gpuvis_graph.cpp:405:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_events, win, _1 );
                                    ^~~~
                                    find
src/gpuvis_graph.cpp:405:83: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_events, win, _1 );
                                                                                   ^~
src/gpuvis_graph.cpp:405:83: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_events, win, _1 );
                                                                                   ^~
                                                                                   y1
In file included from src/gpuvis_utils.cpp:41:0:
src/gpuvis_utils.h:35:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_1;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:36:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_2;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:37:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_3;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:38:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_4;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:39:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_5;
            ^~~~~~~~~~~~
---- src/ya_getopt.c ----
---- src/hash_fnv.c ----
In file included from src/gpuvis.cpp:46:0:
src/gpuvis_utils.h:35:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_1;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:36:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_2;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:37:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_3;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:38:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_4;
            ^~~~~~~~~~~~
src/gpuvis_utils.h:39:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_5;
            ^~~~~~~~~~~~
make: *** [Makefile:143: _release/src/gpuvis_utils.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:143: _release/src/gpuvis_graph.o] Error 1
src/gpuvis.cpp: In static member function ‘static int TraceLoader::thread_func(void*)’:
src/gpuvis.cpp:663:35: error: ‘bind’ is not a member of ‘std’
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                   ^~~~
src/gpuvis.cpp:663:35: note: suggested alternative: ‘find’
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                   ^~~~
                                   find
src/gpuvis.cpp:663:63: error: ‘_1’ was not declared in this scope
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                                               ^~
src/gpuvis.cpp:663:63: note: suggested alternative: ‘y1’
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                                               ^~
                                                               y1
src/gpuvis.cpp:663:67: error: ‘_2’ was not declared in this scope
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                                                   ^~
src/gpuvis.cpp:663:67: note: suggested alternative: ‘._92’
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                                                   ^~
                                                                   ._92
src/gpuvis.cpp: In member function ‘void TraceLoader::init(int, char**)’:
src/gpuvis.cpp:684:32: error: ‘bind’ is not a member of ‘std’
     io.IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                ^~~~
src/gpuvis.cpp:684:32: note: suggested alternative: ‘find’
     io.IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                ^~~~
                                find
src/gpuvis.cpp:684:90: error: ‘_1’ was not declared in this scope
 .IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                                                                    ^~
src/gpuvis.cpp:684:90: note: suggested alternative: ‘y1’
 .IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                                                                    ^~
                                                                                          y1
src/gpuvis.cpp:684:94: error: ‘_2’ was not declared in this scope
 .IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                                                                        ^~
src/gpuvis.cpp:684:94: note: suggested alternative: ‘._92’
 .IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                                                                        ^~
                                                                                              ._92
src/gpuvis.cpp:685:32: error: ‘bind’ is not a member of ‘std’
     io.IniSaveSettingCB = std::bind( imgui_ini_save_settings_cb, _1, _2 );
                                ^~~~
src/gpuvis.cpp:685:32: note: suggested alternative: ‘find’
     io.IniSaveSettingCB = std::bind( imgui_ini_save_settings_cb, _1, _2 );
                                ^~~~
                                find
src/gpuvis.cpp: In member function ‘const std::vector<unsigned int>* TraceEvents::get_tdopexpr_locs(const char*, std::__cxx11::string*)’:
src/gpuvis.cpp:1361:47: error: ‘bind’ is not a member of ‘std’
         tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                               ^~~~
src/gpuvis.cpp:1361:47: note: suggested alternative: ‘find’
         tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                               ^~~~
                                               find
src/gpuvis.cpp:1361:86: error: ‘_1’ was not declared in this scope
       tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                                                                    ^~
src/gpuvis.cpp:1361:86: note: suggested alternative: ‘y1’
       tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                                                                    ^~
                                                                                      y1
src/gpuvis.cpp:1361:90: error: ‘_2’ was not declared in this scope
       tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                                                                        ^~
src/gpuvis.cpp:1361:90: note: suggested alternative: ‘._92’
       tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                                                                        ^~
                                                                                          ._92
src/gpuvis.cpp:1376:61: error: ‘bind’ is not a member of ‘std’
                 tdop_get_keyval_func get_keyval_func = std::bind( filter_get_keyval_func, &event, _1, _2 );
                                                             ^~~~
src/gpuvis.cpp:1376:61: note: suggested alternative: ‘find’
                 tdop_get_keyval_func get_keyval_func = std::bind( filter_get_keyval_func, &event, _1, _2 );
                                                             ^~~~
                                                             find
src/gpuvis.cpp: In member function ‘bool TraceWin::render()’:
src/gpuvis.cpp:1957:55: error: ‘bind’ is not a member of ‘std’
                 tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                       ^~~~
src/gpuvis.cpp:1957:55: note: suggested alternative: ‘find’
                 tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                       ^~~~
                                                       find
src/gpuvis.cpp:1957:109: error: ‘_1’ was not declared in this scope
 key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                                                    ^~
src/gpuvis.cpp:1957:109: note: suggested alternative: ‘y1’
 key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                                                    ^~
                                                                                                             y1
src/gpuvis.cpp:1957:113: error: ‘_2’ was not declared in this scope
 key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                                                        ^~
src/gpuvis.cpp:1957:113: note: suggested alternative: ‘._92’
 key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                                                        ^~
                                                                                                                 ._92
src/gpuvis.cpp:1966:69: error: ‘bind’ is not a member of ‘std’
                         tdop_get_keyval_func get_keyval_func = std::bind( filter_get_keyval_func, &event, _1, _2 );
                                                                     ^~~~
src/gpuvis.cpp:1966:69: note: suggested alternative: ‘find’
                         tdop_get_keyval_func get_keyval_func = std::bind( filter_get_keyval_func, &event, _1, _2 );
                                                                     ^~~~
                                                                     find
make: *** [Makefile:143: _release/src/gpuvis.o] Error 1
✗ fratti@archbook gpuvis $ mkdir build
✓ fratti@archbook gpuvis $ cd build/
✓ fratti@archbook build $ cmake ..
-- The C compiler identification is GNU 7.1.1
-- The CXX compiler identification is GNU 7.1.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
<FindSDL2.cmake>
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
</FindSDL2.cmake>
-- Found SDL2: /usr/lib/libSDL2main.a;/usr/lib/libSDL2.so;-lpthread  
-- Found Freetype: /usr/lib/libfreetype.so (found version "2.8.0") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'gtk+-3.0'
--   Found gtk+-3.0, version 3.22.15

CMAKE_C_FLAGS:  -DUSE_FREETYPE -static-libstdc++ -static-libgcc
CMAKE_CXX_FLAGS:  -DUSE_FREETYPE -static-libstdc++ -static-libgcc -std=c++11
CMAKE_EXE_LINKER_FLAGS: 
CMAKE_MODULE_LINKER_FLAGS: 
CMAKE_SHARED_LINKER_FLAGS: 
CMAKE_STATIC_LINKER_FLAGS: 

-- Configuring done
-- Generating done
-- Build files have been written to: /home/fratti/Projekte/gpuvis/build
✓ fratti@archbook build $ make -j4
Scanning dependencies of target gpuvis
[  5%] Building CXX object CMakeFiles/gpuvis.dir/src/gpuvis_utils.cpp.o
[ 11%] Building CXX object CMakeFiles/gpuvis.dir/src/gpuvis.cpp.o
[ 16%] Building CXX object CMakeFiles/gpuvis.dir/src/tdopexpr.cpp.o
[ 22%] Building CXX object CMakeFiles/gpuvis.dir/src/gpuvis_graph.cpp.o
In file included from /home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:44:0:
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:35:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_1;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:36:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_2;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:37:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_3;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:38:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_4;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:39:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_5;
            ^~~~~~~~~~~~
In file included from /home/fratti/Projekte/gpuvis/src/gpuvis_utils.cpp:41:0:
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:35:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_1;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:36:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_2;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:37:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_3;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:38:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_4;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:39:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_5;
            ^~~~~~~~~~~~
In file included from /home/fratti/Projekte/gpuvis/src/gpuvis.cpp:46:0:
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:35:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_1;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:36:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_2;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:37:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_3;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:38:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_4;
            ^~~~~~~~~~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_utils.h:39:12: error: ‘std::placeholders’ has not been declared
 using std::placeholders::_5;
            ^~~~~~~~~~~~
[ 27%] Building C object CMakeFiles/gpuvis.dir/src/ya_getopt.c.o
[ 33%] Building C object CMakeFiles/gpuvis.dir/src/hash_fnv.c.o
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp: In member function ‘void graph_info_t::init_row_info(TraceWin*, const std::vector<GraphRows::graph_rows_info_t>&)’:
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:385:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_print_timeline, win, _1 );
                                    ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:385:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_print_timeline, win, _1 );
                                    ^~~~
                                    find
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:385:87: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_print_timeline, win, _1 );
                                                                                       ^~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:385:87: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_print_timeline, win, _1 );
                                                                                       ^~
                                                                                       y1
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:390:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_plot, win, _1 );
                                    ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:390:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_plot, win, _1 );
                                    ^~~~
                                    find
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:390:77: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_plot, win, _1 );
                                                                             ^~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:390:77: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_plot, win, _1 );
                                                                             ^~
                                                                             y1
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:395:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_timeline, win, _1 );
                                    ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:395:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_timeline, win, _1 );
                                    ^~~~
                                    find
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:395:85: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_timeline, win, _1 );
                                                                                     ^~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:395:85: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_timeline, win, _1 );
                                                                                     ^~
                                                                                     y1
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:400:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_hw_row_timeline, win, _1 );
                                    ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:400:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_hw_row_timeline, win, _1 );
                                    ^~~~
                                    find
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:400:88: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_hw_row_timeline, win, _1 );
                                                                                        ^~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:400:88: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_hw_row_timeline, win, _1 );
                                                                                        ^~
                                                                                        y1
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:405:36: error: ‘bind’ is not a member of ‘std’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_events, win, _1 );
                                    ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:405:36: note: suggested alternative: ‘find’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_events, win, _1 );
                                    ^~~~
                                    find
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:405:83: error: ‘_1’ was not declared in this scope
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_events, win, _1 );
                                                                                   ^~
/home/fratti/Projekte/gpuvis/src/gpuvis_graph.cpp:405:83: note: suggested alternative: ‘y1’
             rinfo.render_cb = std::bind( &TraceWin::graph_render_row_events, win, _1 );
                                                                                   ^~
                                                                                   y1
[ 38%] Building CXX object CMakeFiles/gpuvis.dir/src/stlini.cpp.o
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp: In static member function ‘static int TraceLoader::thread_func(void*)’:
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:663:35: error: ‘bind’ is not a member of ‘std’
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                   ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:663:35: note: suggested alternative: ‘find’
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                   ^~~~
                                   find
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:663:63: error: ‘_1’ was not declared in this scope
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                                               ^~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:663:63: note: suggested alternative: ‘y1’
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                                               ^~
                                                               y1
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:663:67: error: ‘_2’ was not declared in this scope
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                                                   ^~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:663:67: note: suggested alternative: ‘._92’
     EventCallback trace_cb = std::bind( new_event_cb, loader, _1, _2 );
                                                                   ^~
                                                                   ._92
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp: In member function ‘void TraceLoader::init(int, char**)’:
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:684:32: error: ‘bind’ is not a member of ‘std’
     io.IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:684:32: note: suggested alternative: ‘find’
     io.IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                ^~~~
                                find
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:684:90: error: ‘_1’ was not declared in this scope
 .IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                                                                    ^~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:684:90: note: suggested alternative: ‘y1’
 .IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                                                                    ^~
                                                                                          y1
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:684:94: error: ‘_2’ was not declared in this scope
 .IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                                                                        ^~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:684:94: note: suggested alternative: ‘._92’
 .IniLoadSettingCB = std::bind( imgui_ini_load_settings_cb, &m_imguiwindow_entries, _1, _2 );
                                                                                        ^~
                                                                                              ._92
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:685:32: error: ‘bind’ is not a member of ‘std’
     io.IniSaveSettingCB = std::bind( imgui_ini_save_settings_cb, _1, _2 );
                                ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:685:32: note: suggested alternative: ‘find’
     io.IniSaveSettingCB = std::bind( imgui_ini_save_settings_cb, _1, _2 );
                                ^~~~
                                find
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp: In member function ‘const std::vector<unsigned int>* TraceEvents::get_tdopexpr_locs(const char*, std::__cxx11::string*)’:
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1361:47: error: ‘bind’ is not a member of ‘std’
         tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                               ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1361:47: note: suggested alternative: ‘find’
         tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                               ^~~~
                                               find
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1361:86: error: ‘_1’ was not declared in this scope
       tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                                                                    ^~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1361:86: note: suggested alternative: ‘y1’
       tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                                                                    ^~
                                                                                      y1
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1361:90: error: ‘_2’ was not declared in this scope
       tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                                                                        ^~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1361:90: note: suggested alternative: ‘._92’
       tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_strpool, _1, _2 );
                                                                                        ^~
                                                                                          ._92
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1376:61: error: ‘bind’ is not a member of ‘std’
                 tdop_get_keyval_func get_keyval_func = std::bind( filter_get_keyval_func, &event, _1, _2 );
                                                             ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1376:61: note: suggested alternative: ‘find’
                 tdop_get_keyval_func get_keyval_func = std::bind( filter_get_keyval_func, &event, _1, _2 );
                                                             ^~~~
                                                             find
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp: In member function ‘bool TraceWin::render()’:
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1957:55: error: ‘bind’ is not a member of ‘std’
                 tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                       ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1957:55: note: suggested alternative: ‘find’
                 tdop_get_key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                       ^~~~
                                                       find
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1957:109: error: ‘_1’ was not declared in this scope
 key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                                                    ^~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1957:109: note: suggested alternative: ‘y1’
 key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                                                    ^~
                                                                                                             y1
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1957:113: error: ‘_2’ was not declared in this scope
 key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                                                        ^~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1957:113: note: suggested alternative: ‘._92’
 key_func get_key_func = std::bind( filter_get_key_func, &m_trace_events.m_strpool, _1, _2 );
                                                                                        ^~
                                                                                                                 ._92
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1966:69: error: ‘bind’ is not a member of ‘std’
                         tdop_get_keyval_func get_keyval_func = std::bind( filter_get_keyval_func, &event, _1, _2 );
                                                                     ^~~~
/home/fratti/Projekte/gpuvis/src/gpuvis.cpp:1966:69: note: suggested alternative: ‘find’
                         tdop_get_keyval_func get_keyval_func = std::bind( filter_get_keyval_func, &event, _1, _2 );
                                                                     ^~~~
                                                                     find
make[2]: *** [CMakeFiles/gpuvis.dir/build.make:111: CMakeFiles/gpuvis.dir/src/gpuvis_utils.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/gpuvis.dir/build.make:87: CMakeFiles/gpuvis.dir/src/gpuvis_graph.cpp.o] Error 1
make[2]: *** [CMakeFiles/gpuvis.dir/build.make:63: CMakeFiles/gpuvis.dir/src/gpuvis.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:68: CMakeFiles/gpuvis.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

--- a/src/gpuvis.cpp
+++ b/src/gpuvis.cpp
@@ -27,6 +27,7 @@
 #include <future>
 #include <set>
 #include <unordered_map>
+#include <functional>
 
 #define SDL_MAIN_HANDLED
 #include <SDL.h>

--- a/src/gpuvis_graph.cpp
+++ b/src/gpuvis_graph.cpp
@@ -33,6 +33,7 @@
 #include <vector>
 #include <array>
 #include <limits.h>
+#include <functional>
 
 #include <SDL.h>
 

--- a/src/gpuvis_utils.cpp
+++ b/src/gpuvis_utils.cpp
@@ -30,6 +30,7 @@
 #include <cctype>
 #include <sstream>
 #include <unordered_map>
+#include <functional>
 
 #include <SDL.h>
 

